testing-bundle-structure.py

```diff
@@ -0,0 +1,380 @@
+        else:
+            self.successes["Bundle Folder Contents"].append("There is a single earthmover.yaml file")
+
+        if folder_contents.count("lightbeam.yaml") > 1:
+            self.warnings["Bundle Folder Contents"].append(f"Please make sure there is one lightbeam.yaml file in your bundle, found {folder_contents.count('lightbeam.yaml')}")
+        else:
+            self.successes["Bundle Folder Contents"].append("There is a single lightbeam.yaml file")
+
+
+        self.has_sample_data = False
+
+        if "data" in folder_contents:
+            data_folder = [f for f in os.listdir(f"{self.folder_path}/data")
+                            if f != ".gitkeep"]
+
+        if data_folder:
+            self.has_sample_data = True
+            self.successes["Bundle Folder Contents"].append("Found the following sample data files:\n          " +
+                                                            "\n          ".join(file for file in data_folder))
+        
+        if not self.has_sample_data:
+            self.warnings["Bundle Folder Contents"].append("Missing a sample data file")
+
+
+        self.not_run_tests.remove("Bundle Folder Contents")
+
+
+    # TODO: parsing readme 
+
+    def getting_params(self):
+
+        self.em_params = None
+
+        readme_file_path = self.folder_path + "/README.md"
+
+        try:
+            with open(readme_file_path, "r", encoding="utf-8") as f:
+                readme_file = f.read()
+        except FileNotFoundError:
+            raise NoReadme
+
+        for pattern in ["./earthmover.yaml", "earthmover.yaml"]:
+            try:
+                params_string = readme_file.split(f"earthmover run -c {pattern} -p '")[1].split("'\n```")[0]
+                self.successes["Getting EM Parameters"].append("Found sample Earthmover run command in the README.md")
+                break
+            except IndexError:
+                continue
+        else:
+            self.warnings["Getting EM Parameters"].append("Please make sure there is a sample Earthmover run command in README.md")
+
+
+
+        # FIXME: maybe better error handling for this
+        try:
+            em_params_dict = json.loads(params_string)
+            self.em_params = json.dumps(em_params_dict)
+            self.successes["Getting EM Parameters"].append("The Earthmover command is valid JSON")
+        except json.JSONDecodeError:
+            self.warnings["Getting EM Parameters"].append("Unable to use the provided earthmover command, invalid JSON: ", params_string) 
+
+        self.not_run_tests.remove("Getting EM Parameters")
+
+
+
+
+    # TODO: compiling yaml 
+
+    def compiling_yaml(self):
+
+        if self.em_params is None:
+            raise NoEMParams
+
+        self.em_yaml_path = self.folder_path + "/earthmover.yaml"
+
+        command = f'''
+        source ~/.venv/earthmover/bin/activate && \
+        earthmover compile -c {self.em_yaml_path} -p '{self.em_params}' 
+        '''
```

Comment:
Definitely can't guarantee that the user has a virtualenv named earthmover. Better just to run the command and let the `no such command: earthmover` error surface up to the user